src/pixel_formats.rs

```diff
@@ -0,0 +1,308 @@
+//! get the color value to the least significant bit (that is, how many right shifts are needed).
+//!
+//! For example, consider the 32-bit pixel value 0x01020304 with a depth of 24. Let's say the pixel format has a red shift of
+//! 0, green shift of 8, blue shift of 16, all colors have a max value of 255, and the host is
+//! little-endian. This is the pixel format little-endian xBGR.
+//!
+//! So to get the value of each color, we would do:
+//! - red = (0x01020304 >> 0) & 255 = 0x04
+//! - blue = (0x01020304 >> 8) & 255 = 0x03
+//! - green = (0x01020304 >> 16) & 255 = 0x02
+//!
+//! This is relatively straightforward when considering a single pixel format. But an RFB server
+//! must be able to translate between pixel formats, including translating between hosts of
+//! different endianness. Further, it is convenient to represent pixels using a vector of bytes instead
+//! of a vector of n-bit values, and transformations on pixels
+//! are done for groups of bytes representing a pixel, rather than a single pixel value. But
+//! thinking about pixels in this representation can be tricky, as the pixel format describes
+//! shifts, which operate the same on a value regardless of endianness, but code operating on a
+//! byte vector must be endian-aware.
+//!
+//! If we think about the same pixel before as a byte vector, we would have the following
+//! representation: [ 0x04, 0x03, 0x02, 0x01 ]. Note that the bytes are in reverse order from the
+//! value above because the host is little-endian, so the least-significant byte (0x04) is first.
+//!
+//! So to get the value of each color, we would do:
+//! - red = pixel\[0\] & 255 = 0x04
+//! - green = pixel\[1\] & 255 = 0x03
+//! - blue = pixel\[2\] & 255 = 0x02
+//!
+//! If the same byte vector were from an RGBx big endian pixel, the color values would be:
+//! - red = pixel\[3\] & 255 = 0x01
+//! - green = pixel\[2\] & 255 = 0x02
+//! - blue = pixel\[1\] & 255 = 0x03
+//!
+
+///  Utility functions and constants related to fourcc codes.
+///
+/// Fourcc is a 4-byte ASCII code representing a pixel format. For example, the value
+/// 0x34325258 is '42RX' in ASCII (34='4', 32='2', 52='R', and 58='X'). This code maps to the pixel format 32-bit little-endian xRGB.
+///
+/// A good reference for mapping common fourcc codes to their corresponding pixel formats is the drm_fourcc.h header file in the
+/// linux source code.
+pub mod fourcc {
+    use super::rgb_888;
+    use crate::rfb::{ColorFormat, ColorSpecification, PixelFormat};
+    use anyhow::{anyhow, Result};
+
+    // XXX: it might make sense to turn fourcc values into a type (such as an enum or collection of
+    // enums)
+    pub const FOURCC_XR24: u32 = 0x34325258; // little-endian xRGB, 8:8:8:8
+    pub const FOURCC_RX24: u32 = 0x34325852; // little-endian RGBx, 8:8:8:8
+    pub const FOURCC_BX24: u32 = 0x34325842; // little-endian BGRx, 8:8:8:8
```

Comment:
Missing `XB24`/`XBGR8888`?

Reply:
I'm missing a bunch of codes actually, haha. I included the ones I did because XR24 was the edk2 default and the other two were mostly for testing. Was there are a reason you asked about that one in particular?

Reply:
Oh, it just seemed like that was the only missing X/R/G/B 8:8:8:8 format that was missing. https://github.com/torvalds/linux/blob/e35e5b6f695d241ffb1d223207da58a1fbcdff4b/include/uapi/drm/drm_fourcc.h#L156-L160

Reply:
Ah, I see you meant an xRGB format specifically. I've added that one.